src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,98 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
- * реалізації ArrayList (default capacity, newCapacity...)</p>
- */
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEF_CAPACITY = 10;
+    private Object[] element;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        element = new Object[DEF_CAPACITY];
+    }
+
+    private void resize(int minCapacity) {
+        if (minCapacity > element.length) {
+            int newCapacity = (minCapacity * 3) / 2 + 1;
+            Object[] oldData = element;
+            element = (T[]) new Object[newCapacity];
+            System.arraycopy(oldData, 0, element, 0, size);
+        }
+    }
+
+    private void realIndex(int index) {
+        if (index == size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index");
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        resize(size + 1);
```

Comment:
why is logic of checking capacity different in the two `add()` methods?
(see lines 32 and 38)

Reply:
fix